### PR TITLE
Voronoi-Otsu labeling + instance segmentation code overhaul

### PR DESCRIPTION
**Pull request by C-Achard**
- Adds Voronoi-Otsu labeling from pyclesperanto-prototype
- Improves classes and widgets for instance segmentation

Note to self : merge with main after #28 has been merged

### CODE REVIEW

**Review by MMathisLab — APPROVED** [review]

File: napari_cellseg3d/code_models/model_instance_seg.py [hunk 1]
```diff
@@ -1,24 +1,83 @@
...
+            num_sliders: Number of Slider UI elements needed to set the parameters of the function
+            num_counters: Number of DoubleIncrementCounter UI elements needed to set the parameters of the function
+            widget_parent: parent for the declared widgets
+
+        """
+        self.name = name
+        self.function = function
+        self.counters: List[ui.DoubleIncrementCounter] = []
+        self.sliders: List[ui.Slider] = []
+        if num_sliders > 0:
```
Comment (CYHSM) [inline napari_cellseg3d/code_models/model_instance_seg.py:51]: Might be a tad bit cleaner if these two are in separate methods setup_sliders and setup_counters, or in one method setup_widgets(num_sliders, num_counters)

File: napari_cellseg3d/code_models/model_instance_seg.py [hunk 2]
```diff
@@ -50,18 +109,48 @@ def get_dict(self):
 
 
 def threshold(volume, thresh):
+    """Remove all values smaller than the specified threshold in the volume"""
     im = np.squeeze(volume)
     binary = im > thresh
     return np.where(binary, im, np.zeros_like(im))
 
 
+def voronoi_otsu(
```
Comment (CYHSM) [inline napari_cellseg3d/code_models/model_instance_seg.py:118]: Might be helpful for some people to include a link to the code for the Voronoi Otsu method in the doc string:
https://github.com/clEsperanto/pyclesperanto_prototype/blob/master/pyclesperanto_prototype/_tier9/_voronoi_otsu_labeling.py

File: napari_cellseg3d/dev_scripts/correct_labels.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+from tifffile import imread, imwrite
+from tqdm import tqdm
+
+import napari_cellseg3d.dev_scripts.artefact_labeling as make_artefact_labels
+from napari_cellseg3d.code_models.model_instance_seg import binary_watershed
+
+# import sys
+# sys.path.append(str(Path(__file__) / "../../"))
+"""
+New code by Yves Paychère
```
Comment (CYHSM) [inline napari_cellseg3d/dev_scripts/correct_labels.py:20]: Should there be a link from where this comes from?

**Review by CYHSM — COMMENTED** [review]
Just some minor comments

File: napari_cellseg3d/code_models/model_instance_seg.py (hunk 1 above)
Comment (C-Achard) [reply napari_cellseg3d/code_models/model_instance_seg.py:51]: Fixed in #36

File: napari_cellseg3d/code_models/model_instance_seg.py (hunk 2 above)
Comment (C-Achard) [reply napari_cellseg3d/code_models/model_instance_seg.py:118]: Solved in #36

File: napari_cellseg3d/dev_scripts/correct_labels.py (hunk 3 above)
Comment (C-Achard) [reply napari_cellseg3d/dev_scripts/correct_labels.py:20]: This is code from the MLCourse project of last semester; so it was written specifically for this :) I can link to my repo where the original code lives, though ?

**Pull request merged**